### PR TITLE
Stream deletion API

### PR DESCRIPTION
**Pull request by Jmgr**
This PR adds an RPC so that clients can ask for the deletion of a stream. All corresponding partitions will also be removed, as well as their persistent storage.

Note that Liftbridge clients that have subscribed to a stream that will get deleted do not receive any notification.

A protobuf generated file has been added in a separate commit since it is suspected that a different version of protoc is being used, so those changes may or may not be kept.

### CODE REVIEW

File: api.proto [hunk 1]
```diff
@@ -17,6 +17,17 @@ message CreateStreamResponse {
     // Intentionally empty.
 }
 
+// DeleteStreamRequest is send to delete a stream.
```
Comment (cruickshankpg) [inline api.proto]: is send => is sent

File: api.proto [hunk 2]
```diff
@@ -134,6 +145,9 @@ service API {
     // already exists.
     rpc CreateStream(CreateStreamRequest) returns (CreateStreamResponse) {}
 
+    // TODO
```
Comment (cruickshankpg) [inline api.proto]: TODO

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module main
```
Comment (cruickshankpg) [inline go.mod]: should you be adding this?

Comment (Jmgr) [reply go.mod]: For now, yes. This is required to get "replace" instructions to work in other projects that depend on this.

File: api.proto (hunk 2 above)
Comment (Jmgr) [reply api.proto]: :+1:

File: api.proto (hunk 1 above)
Comment (Jmgr) [reply api.proto]: :+1:

**Comment by Jmgr** [comment]
Merged into upstream, closing.

**Pull request closed without merging**